### DISCUSSION TITLE
Member not found issue when dotnet publish Browser project (wasm)

### DISCUSSION THREAD

**Discussion by eugenca** [AvaloniaUI/Avalonia#14990] [Q&A] [answered]
### Describe the bug

I get this error when dotnet publish only in Browser target only (wasm)
Error happens in library in both Release and Debug, only in dotnet publish

```
Error  Message = FunctionParser::Parse Error when parsing expression x. ErrorMessage: x is not a member of type SolverLibrary.SolverMath.Solvers.Context (Parameter 'propertyOrFieldName), stack:    at System.Linq.Expressions.Expression.PropertyOrField(Expression , String )
at NCalc.LambdaExpressionVistor.Visit(Identifier )
at NCalc.Domain.Identifier.Accept(LogicalExpressionVisitor )
at SolverLibrary.FunctionParser.Parse[Context](String funcString)
trace=at SolverLibrary.FunctionParser.Parse[Context](String funcString)
```
When building app with visual studio, everything is fine, but there are a lot more of files in AppBundle
I cannot understand why parameter is not exposed

Context struct looks like this
```
public struct Context
{
    public double x { get { return Params[0]; } }
    public double[] Params;
}
```

### To Reproduce

Not sure how to reproduce

### Avalonia version

11.0.6

### OS

WebAssembly

### Additional context

![image](https://github.com/AvaloniaUI/Avalonia/assets/5236164/5d916121-b1a4-4db3-a20d-a191061a81a4)

**Comment by maxkatz6** [comment]
Looks like a trimming issue. 
Please check out https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trim-self-contained

**Comment by eugenca** [comment]
running  ```dotnet publish -c Debug -p:PublishTrimmed=false```
makes build working, but its practically identical to dotnet build

**Reply by eugenca** [reply]
```
[RequiresUnreferencedCode(ExpressionRequiresUnreferencedCode)]
public static MemberExpression PropertyOrField(Expression expression, string propertyOrFieldName)
```

PropertyOrField has warning attribute about this actually

**Reply by maxkatz6** [reply, marked as answer]
Yes, this code is not safe for trimming nor AOT, so you can't get the best performance/size results in Browser. But you can try different options with "root descriptors" or "DynamicDependency" as it is mentioned on that documentation by MS.

**Reply by eugenca** [reply]
Adding 
```
<ItemGroup>
	<TrimmerRootAssembly Include="SolverLibrary" />
</ItemGroup>
```
to Browser project file solved the issue!
This library depends on NCalc library that uses PropertyOrField method.

Thanks for direction on the source of the problem, Ill continue with that approach if it would work!

**Reply by eugenca** [reply]
For comparison - trimmed release - 31MB, untrimmed release - 65MB, trimmed with root - 32 MB
